### PR TITLE
Tests/scaffold fsm all

### PR DESCRIPTION
**Pull request by Karrenbelt**
## Proposed changes

Increasing scaffold test coverage to all valory `fsm_specification.yaml`'s

## Fixes

If it fixes a bug or resolves a feature request, be sure to link to that issue.

## Types of changes

What types of changes does your code introduce? (A **breaking change** is a fix or feature that would cause existing functionality and APIs to not work as expected.)
_Put an `x` in the box that applies_

- [ ] Non-breaking fix (non-breaking change which fixes an issue)
- [ ] Breaking fix (breaking change which fixes an issue)
- [ ] Non-breaking feature (non-breaking change which adds functionality)
- [ ] Breaking feature (breaking change which adds functionality)
- [ ] Refactor (non-breaking change which changes implementation)
- [ ] Messy (mixture of the above - requires explanation!)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../blob/main/CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `main` branch (left side). Also you should start your branch off our `main`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have locally run services that could be impacted and they do not present failures derived from my changes

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 1]
```diff
@@ -18,45 +18,46 @@
...
-    def test_run(
-        self,
-    ) -> None:
+    @pytest.mark.parametrize("fsm_spec_file", fsm_specifications)
+    def test_run(self, fsm_spec_file: Path) -> None:
         """Test run."""
-        self.set_agent_context(self.agent_name)
-        path_to_spec_file = Path(ROOT_DIR) / self.fsm_spec_file
+        my_skill = f"test_{fsm_spec_file.parts[-2]}"
+        self.cli_options[-3] = my_skill
```
Comment (Karrenbelt) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:59]: local variable for the reader

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 2]
```diff
@@ -18,45 +18,46 @@
...
 # trigger population of autonomy commands
 import autonomy.cli.core  # noqa
 
+from packages.valory import skills
+
 from tests.conftest import ROOT_DIR
 
 
+VALORY_SKILLS_PATH = Path(os.path.join(*skills.__package__.split("."))).absolute()
+fsm_specifications = VALORY_SKILLS_PATH.glob("**/fsm_specification.yaml")
```
Comment (Adamantios) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:39]: Nice, fewer chances to get out of sync.

File: tests/test_autonomy/test_cli/test_scaffold_fsm.py [hunk 3]
```diff
@@ -18,45 +18,46 @@
...
+    def test_run(self, fsm_spec_file: Path) -> None:
         """Test run."""
-        self.set_agent_context(self.agent_name)
-        path_to_spec_file = Path(ROOT_DIR) / self.fsm_spec_file
+        my_skill = f"test_{fsm_spec_file.parts[-2]}"
+        self.cli_options[-3] = my_skill
+        path_to_spec_file = Path(ROOT_DIR) / fsm_spec_file
         args = [*self.cli_options, path_to_spec_file]
         result = self.run_cli_command(*args, cwd=self._get_cwd())
         assert result.exit_code == 0
```
Comment (Adamantios) [inline tests/test_autonomy/test_cli/test_scaffold_fsm.py:63]: I wonder if it is enough to check for the exit code only.

**Review by Adamantios — COMMENTED** [review]
Do we discard the scaffolded test skills after running the test? If not, let's rm them on finalization.

**Review by Adamantios — APPROVED** [review]

**Pull request merged**